server/apps/archive/archive_spike.py

```diff
@@ -149,15 +141,17 @@ def get_unspike_updates(self, doc):
                    ITEM_OPERATION: ITEM_UNSPIKE}
 
         desk_id = doc.get('task', {}).get('desk')
+        stage_id = None
         if desk_id:
             desk = app.data.find_one('desks', None, _id=desk_id)
+            stage_id = desk['incoming_stage']
             updates['task'] = {
-                'desk': str(desk_id),
-                'stage': str(desk['incoming_stage']) if desk_id else None,
+                'desk': desk_id,
+                'stage': stage_id if desk_id else None,
                 'user': None
             }
 
-        updates['expiry'] = get_expiry(desk_id=desk_id)
+        updates['expiry'] = get_expiry(desk_id=desk_id, stage_id=stage_id)
```

Comment:
It's already there in line 115 right? I was pointing to that constant only. But yes agreed, it can be part of some global, may item metadata.